frontend/src/views/account-mgmt/FacilityView.vue

```diff
@@ -0,0 +1,271 @@
+              </v-row>
+            </div>
+          </v-skeleton-loader>
+        </v-card>
+      </v-col>
+    </v-row>
+    <v-row>
+      <v-col class="ml-6 pr-9 pt-0 pb-0">
+        <h4>Primary Contact</h4>
+      </v-col>
+    </v-row>
+    <v-row>
+      <v-col cols="12" class="ml-6 pr-9 pt-0">
+        <v-card class="pa-6 mb-3" variant="outlined">
+          <v-skeleton-loader :loading="loading" type="table-tbody">
+            <div class="w-100"> <!-- NOTE: div was required due to dynamic v-row within v-skeleton-loader, otherwise intended row formatting breaks -->
+              <v-row>
+                <v-col>
+                  <AppLabel>Current primary contact info:</AppLabel>
+                </v-col>
+                <v-col>
+                  <v-row no-gutters justify="end">
+                    <v-icon v-if="isEditablePrimaryContact" icon="fa:fa-regular fa-edit" class="mr-4"></v-icon>
+                  </v-row>
+                </v-col>
+              </v-row>
+              <v-row v-for="(item, index) in this.primaryContact" :key="index">
+                <v-col cols="auto">
+                  <AppLabel>Name: </AppLabel>
+                </v-col>
+                <v-col cols="2">
+                  {{ item.firstName }} {{ item.lastName }}
+                </v-col>
+                <v-col cols="auto">
+                  <AppLabel>Role:</AppLabel>
+                </v-col>
+                <v-col cols="auto">
+                  {{ getRoleNameById(item.role) }}
+                </v-col>
+              </v-row>
+            </div>
+          </v-skeleton-loader>
+        </v-card>
+      </v-col>
+    </v-row>
+    <v-row>
+      <v-col cols="10">
+        <v-row>
+          <AppButton id="back-to-account-mgmt" :primary="false" size="medium" width="400px" :to="{ name: 'account-mgmt' }" :loading="loading">&larr; Back to Account Management</AppButton>
+        </v-row>
+      </v-col>
+      <v-col cols="2">
+        <v-row>
+          <AppButton id="cancel" :primary="true" size="large" :to="{ name: 'organization-facility' }" :loading="loading" class="mr-6">Cancel</AppButton>
+          <AppButton id="save" :disabled="true" :primary="true" size="large" :to="{ name: 'account-mgmt' }" :loading="loading">Save</AppButton>
+        </v-row>
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+
+<script>
+import AppButton from '@/components/ui/AppButton.vue'
+import AppLabel from '@/components/ui/AppLabel.vue'
+import FacilityInfo from '@/components/facilities/FacilityInfo.vue'
+import FacilityService from '@/services/facilityService'
+import alertMixin from '@/mixins/alertMixin'
+import { useAppStore } from '@/stores/app'
+import { mapState } from 'pinia'
+import { useAuthStore } from '@/stores/auth'
+
+export default {
+  name: 'FacilityView',
+  components: { AppButton, AppLabel, FacilityInfo },
+  mixins: [alertMixin],
+  data() {
+    return {
+      facilityId: null,
+      licences: undefined,
+      contacts: undefined,
```

Comment:
Would an empty array be a better default?

Reply:
fixed.